src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -1169,5 +1213,34 @@ protected boolean isInvalidSample(double sampleRate) {
         return sampleRate != 1 && ThreadLocalRandom.current().nextDouble() > sampleRate;
     }
 
+    protected boolean isOriginDetectionEnabled(String containerID, boolean originDetectionEnabled, boolean hasEntityID) {
+        if (!originDetectionEnabled || hasEntityID || (containerID != null && !containerID.isEmpty())) {
+            // origin detection is explicitly disabled
+            // or DD_ENTITY_ID was found
+            // or a user-defined container ID was provided
+            return false;
+        }
 
+        final String value = System.getenv(ORIGIN_DETECTION_ENABLED_ENV_VAR);
+        if (value != null && !value.trim().isEmpty()) {
```

Comment:
Would it make sense to do the value check below on a trimmed value? Otherwise `<SP>off` would be treated as true.

```suggestion
        String value = System.getenv(ORIGIN_DETECTION_ENABLED_ENV_VAR);
        value = value != null ? value.trim() : null;
        if (value != null && !value.isEmpty()) {
```

Reply:
nice catch!